packages/iris-grid/src/IrisGrid.tsx

```diff
@@ -1812,8 +1814,10 @@ class IrisGrid extends Component<IrisGridProps, IrisGridState> {
         filter: this.makeQuickFilter(column, text, formatter.timeZone),
       });
     });
-
-    this.startLoading('Rebuilding filters...', { resetRanges: true });
+    // If partitionConfig is empty, no need to rebuild filters
+    if (partitionConfig?.mode !== 'empty' && newQuickFilters.size > 0) {
```

Comment:
We can likely just return early from this function if there are no filters, then nothing to rebuild. At the beginning of the function, can just check:
```
if (advancedFilters.size() === 0 && quickFilters.size() === 0) {
  return;
}
```